### PR TITLE
feat: add --order flag plus some refactor

### PR DESCRIPTION
**Pull request by David-Pena**

### CODE REVIEW

File: src/helpers/index.ts [hunk 1]
```diff
@@ -1,5 +1,59 @@
...
+  groupBy: validGroupByOptions,
+  orderBy: validOrderOptions
+}
+
+export function customOptionType<T>(value: T, flag: Flag) {
+  const validOptions = flagOptions[flag];
+
+  if (!validOptions.includes(value as unknown as GroupBy & OrderBy)) {
+    // eslint-disable-next-line node/prefer-global/process
+    process.exit(1)
```
Comment (rrd108) [inline src/helpers/index.ts:32]: I think you should throw an error here. It gives some feedback to humans why it exited.

File: src/helpers/index.ts [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+
+  return value
+}
+
+export function calculateCodeHealth(health: Health[], linesCount: number, filesCount: number): CodeHealthResponse {
+  const { errors, warnings } = health.reduce((acc, { errors, warnings }) => ({ errors: acc.errors + errors, warnings: acc.warnings + warnings }), { errors: 0, warnings: 0 })
+
+  console.log(`Found ${BG_ERR}${Intl.NumberFormat('en-US').format(errors)} errors${BG_RESET}, and ${BG_WARN}${Intl.NumberFormat('en-US').format(warnings)} warnings${BG_RESET}, ${BG_INFO}${Intl.NumberFormat('en-US').format(linesCount)} lines${BG_RESET} of code in ${BG_INFO}${Intl.NumberFormat('en-US').format(filesCount)} files${BG_RESET}`)
+
+  const codeHealth = Math.ceil((1 - (errors * 1.5 + warnings) / linesCount) * 100)
```
Comment (rrd108) [inline src/helpers/index.ts:43]: I know it was written by me but 1.5 is a magic number, it should move out to a constant.

File: src/helpers/index.ts [hunk 3]
```diff
@@ -1,5 +1,59 @@
...
+
+  if (codeHealth < 75) {
```
Comment (rrd108) [inline src/helpers/index.ts:45]: Here I also added magic numbers. Please move them to constants

File: src/rulesReport.ts [hunk 4]
```diff
@@ -98,10 +98,48 @@ export const reportRules = (groupBy: GroupBy) => {
...
+    if (orderBy === 'desc') {
+      return countB - countA
+    }
+
+    return countA - countB
+  })
 
   // Output the report grouped by file
-  Object.keys(offensesGrouped).forEach((key) => {
+  // Object.keys(offensesGrouped).forEach((key) => {
```
Comment (rrd108) [inline src/rulesReport.ts:116]: Do we want to keep all this commented out code?

**Review by rrd108 — CHANGES_REQUESTED** [review]

File: src/rulesReport.ts (hunk 4 above)
Comment (David-Pena) [reply src/rulesReport.ts:116]: My bad 👀 that was while testing the new approach. I'll fix all comments tomorrow 🫡

File: src/helpers/index.ts (hunk 2 above)
Comment (David-Pena) [reply src/helpers/index.ts:43]: Do you use camelCase for constants or uppercase - snake case. e.g ERROR_WEIGHT = 1.5

Comment (rrd108) [reply src/helpers/index.ts:43]: UPPER_CASE

File: src/helpers/index.ts (hunk 1 above)
Comment (David-Pena) [reply src/helpers/index.ts:36]: Added in lastest commit

File: src/helpers/index.ts (hunk 2 above)
Comment (David-Pena) [reply src/helpers/index.ts]: Added in latest commit

File: src/helpers/index.ts (hunk 3 above)
Comment (David-Pena) [reply src/helpers/index.ts]: Added in latest commit

File: src/rulesReport.ts (hunk 4 above)
Comment (David-Pena) [reply src/rulesReport.ts]: Removed in latest commit

**Review by rrd108 — APPROVED** [review]

**Pull request merged**